File: RxTools-library/src/main/java/com/vondear/rxtools/view/wheelhorizontal/ArrayWheelAdapter.java
package com.vondear.rxtools.view.wheelhorizontal;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;

/**
 * @author vondear
 * The simple Array spinnerwheel adapter
 * @param <T> the element type
 */
public class ArrayWheelAdapter<T> extends AbstractWheelTextAdapter {
    
    // items
    private T items[];

    /**
     * Constructor
     * @param context the current context
     * @param items the items
     */
    public ArrayWheelAdapter(Context context, T items[]) {
        super(context);
        
        //setEmptyItemResource(TEXT_VIEW_ITEM_RESOURCE);
        this.items = items;
    }
    
    @Override
    public CharSequence getItemText(int index) {
        if (index >= 0 && index < items.length) {
            T item = items[index];
            if (item instanceof CharSequence) {
                return (CharSequence) item;
            }
            return item.toString();
        }
        return null;
    }

    @Override
    public View getItem(int index, View convertView, ViewGroup parent) {
    	// TODO Auto-generated method stub
    	return super.getItem(index, convertView, parent);
    }
    
    @Override
    public int getItemsCount() {
        return items.length;
    }
}

File: RxUI/src/main/java/com/tamsiree/rxui/view/dialog/wheel/ArrayWheelAdapter.kt
package com.tamsiree.rxui.view.dialog.wheel

import android.content.Context

/**
 * @param <T> the element type
 * The simple Array wheel adapter
 * @author tamsiree
</T> */
open class ArrayWheelAdapter<T>
//setEmptyItemResource(TEXT_VIEW_ITEM_RESOURCE);
(context: Context, // items
 private val items: Array<T>) : AbstractWheelTextAdapter(context, 0, 0) {
    public override fun getItemText(index: Int): CharSequence? {
        if (index >= 0 && index < items.size) {
            val item = items[index]
            return if (item is CharSequence) {
                item
            } else item.toString()
        }
        return null
    }

    override val itemsCount: Int
        get() = items.size

}